src/installer/corehost/corehost.cpp

```diff
@@ -218,6 +218,22 @@ int exe_start(const int argc, const pal::char_t* argv[])
             propagate_error_writer_t propagate_error_writer_to_hostfxr(set_error_writer_fn);
 
             rc = main_fn_v2(argc, argv, host_path_cstr, dotnet_root_cstr, app_path_cstr);
+
+            if (rc == StatusCode::FrameworkMissingFailure)
+            {
+                pal::string_t ver = get_directory(fxr_path);
+                remove_trailing_dir_seperator(&ver);
+
+                fx_ver_t fx_ver;
+                fx_ver_t::parse(get_filename(ver), &fx_ver, false);
+
+                if (fx_ver.get_major() < 3)
```

Comment:
This will work - alternatively we could simply check for the ability to set custom error writer on the `hostfxr` - which is sort of a more precise check (as that is really the functionality we want, we don't care about versions really). I'll leave that up to you.

Reply:
In any case, please add a comment describing what is this trying to achieve and why. The if as is (for the specific error code and so on) feels very arbitrary.